### PR TITLE
Make the awaiter struct and its functionality public.

### PR DESCRIPTION
**Pull request by harry10641**
Reason for it to be reused via code in integration test in services.

### CODE REVIEW

File: api/constants.go [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package awaiter
+
+const Version = "1.1.0"
```
Comment (walles) [inline api/constants.go:3]: What will the process be for keeping this up to date if / when new `await` versions are released?

**Review by walles — APPROVED** [review]
Some issues, but I don't think any new issues will be added by this change so this should be fine.

File: api/constants.go (hunk 1 above)
Comment (harry10641) [reply api/constants.go:3]: I also have no idea why this is kept like this.  Ideally there should be a proper injection of version while building the binary from tag.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      - "v*.*.*"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Run release script
+        run: make rel
```
Comment (walles) [inline .github/workflows/release.yml:15]: The `Makefile` should do something like this to get the version number into the binaries:

`go build -ldflags="-s -w -X main.versionString=${VERSION}" -o "${BINARY}"`

I don't remember what's what in ^, you'll have to figure it out :)

https://github.com/walles/moar/blob/63839bd32535542be355dac14ce6451d90bc4a28/build.sh#L28

File: api/constants.go (hunk 1 above)
Comment (harry10641) [reply api/constants.go]: done

File: .github/workflows/release.yml (hunk 2 above)
Comment (harry10641) [reply .github/workflows/release.yml:15]: Done

File: README.md [hunk 3]
```diff
@@ -49,6 +49,10 @@ or
         	Set more verbose output mode
 
 
+## Release
+
+Push new tag to trigger the new release for await binary.
```
Comment (walles) [inline README.md:54]: Please add an example!

File: Makefile [hunk 4]
```diff
@@ -46,6 +46,7 @@ test:
 	go test -v
 
 .PHONY: rel
+VERSION="$(shell git describe --tags --always --abbrev=0 --match='v[0-9]*.[0-9]*.[0-9]*' 2> /dev/null | sed 's/^.//')"
 rel:
 	GOOS=darwin GOARCH=amd64 go build -o await-darwin-amd64 github.com/betalo-sweden/await
 	GOOS=linux  GOARCH=amd64 go build -o await-linux-amd64  github.com/betalo-sweden/await
```
Comment (walles) [inline Makefile:52]: These need to set the `$VERSION`.

File: Makefile [hunk 5]
```diff
@@ -46,6 +46,7 @@ test:
 	go test -v
 
 .PHONY: rel
+VERSION="$(shell git describe --tags --always --abbrev=0 --match='v[0-9]*.[0-9]*.[0-9]*' 2> /dev/null | sed 's/^.//')"
 rel:
-	GOOS=darwin GOARCH=amd64 go build -o await-darwin-amd64 github.com/betalo-sweden/await
-	GOOS=linux  GOARCH=amd64 go build -o await-linux-amd64  github.com/betalo-sweden/await
+	GOOS=linux  GOARCH=amd64 go build -ldflags="-s -w -X 'github.com/betalo-sweden/await/api.Version=${VERSION}'" -o await-darwin-amd64 github.com/betalo-sweden/await
```
Comment (walles) [inline Makefile:51]:
```suggestion
	GOOS=darwin GOARCH=amd64 go build -ldflags="-s -w -X 'github.com/betalo-sweden/await/api.Version=${VERSION}'" -o await-darwin-amd64 github.com/betalo-sweden/await
```

**Pull request merged**